monkey/infection_monkey/monkey.py

```diff
@@ -180,14 +186,27 @@ def _setup(self):
         control_channel.register_agent(self._opts.parent)
 
         config = control_channel.get_config()
-        self._monkey_inbound_tunnel = self._control_client.create_control_tunnel(
-            config.keep_tunnel_open_time
+
+        local_port = get_free_tcp_port()
+        sock, ip_str, port = connect([self._opts.server])
+        sock.close()
+        user_handler, connection_handler, pipe_spawner = build_tcprelay_deps(
+            local_port,
+            IPv4Address(ip_str),
+            port,
+            client_disconnect_timeout=config.keep_tunnel_open_time,
+        )
+        self._relay_user_handler = user_handler
+        self._relay = TCPRelay(
+            self._relay_user_handler,
+            connection_handler,
+            pipe_spawner,
         )
```

Comment:
Probably all of this should go under a different function and just call that from `_setup`.

Reply:
This code has been shortened. Let me know if we think we still want to change this.